pkg/kubelet/cm/deviceplugin/device_plugin_stub.go

```diff
@@ -86,14 +86,6 @@ func (m *Stub) Start() error {
 	pluginapi.RegisterDevicePluginServer(m.server, m)
 
 	go m.server.Serve(sock)
-	// Wait till grpc server is ready.
-	for i := 0; i < 10; i++ {
-		services := m.server.GetServiceInfo()
-		if len(services) > 0 {
-			break
-		}
-		time.Sleep(1 * time.Second)
-	}
```

Comment:
Should we dial here to wait grpc server is ready?

Reply:
No, no need. Now `WithBlock` in `Dial` will ensure this for us.

Reply:
I think it's actually better to have the Start() call guarantee that the server is ready.
That's a saner pattern and the cost isn't very high compared to the `sleep(10)` we've had for the past 3 monts.

Reply:
1. How would you guarantee that? keeping the `for` loop?
2. What would we gain by this guarantee other than losing some cpu cycles?

Reply:
> How would you guarantee that? keeping the for loop?
```
c, err := dial(socket, timeout)
...
c.Close()
```

Reply:
> What would we gain by this guarantee other than losing some cpu cycles?
 
Code robustness. When you create a stub device plugin you have the guarantee that the gRPC server is online. Thus preventing bugs we are currently solving.